### PR TITLE
STI on Pieces

### PR DESCRIPTION
**Pull request by fistoriza**
first pass on the STI for pieces.

### CODE REVIEW

**Comment by kboygit** [comment]
[Set up Single Table Inheritance on chess pieces.](https://trello.com/c/3Ig7mprN/11-set-up-single-table-inheritance-on-chess-pieces)

**Comment by kboygit** [comment]
Need to be reviewed and merge

**Comment by kboygit** [comment]
Thumbs up good to go check any issues along the way.

**Comment by nnadkarny** [comment]
Thumbs up!

File: app/models/user.rb [hunk 1]
```diff
@@ -3,4 +3,8 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  has_many :games
+  has_many :pieces
```
Comment (ilyakrasnov) [inline app/models/user.rb]: No need for users to have pieces... User-> Game-> Piece already covers the association.

Comment (kirbygit) [inline app/models/user.rb]: oops I see , Piece already covers it piece -> game -> user so all user needs is association with the game?

**Comment by ilyakrasnov** [comment]
Yep, the user already has the association with the game..

**Pull request merged**